lua/draw.lua

```diff
@@ -19,20 +21,122 @@ function draw_ap()
 	)
 
 	print(
-		"ap: " .. curr_ap .. "/" .. max_ap,
+		"ap:" .. curr_ap .. "/" .. max_ap,
 		ap_top_left_x + 1,
 		ap_top_left_y + 1,
 		0
 	)
 end
 
+function draw_capture()
+	local mouse_x = stat(32)
+	local mouse_y = stat(33)
+
+	if capturing then
+		for agent in all(agents) do
+			local top_left_x = agent.x - 1
+			local top_left_y = agent.y - 1
+			local bot_right_x = agent.x + 1
+			local bot_right_y = agent.y + 1
+
+			if top_left_x < min_x or top_left_x == nil then
+				top_left_x = min_x
+			end
+			if top_left_y < min_y or top_left_y == nil
+				then top_left_y = min_y
+				end
+			if bot_right_x > max_x or bot_right_x == nil then
+				bot_right_x = max_x
+			end
+			if bot_right_y > max_y  or bot_right_y == nil then
+				bot_right_y = max_y
+			end
```

Comment:
None of the first four calculations can return nil, so subsequent nil checks are unnecessary